pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -296,3 +388,40 @@ func (r *ReconcileWindowsMachine) validateUserData(privateKey []byte) error {
 	}
 	return nil
 }
+
+// isAllowedDeletion determines if the number of machines after deletion of the given machine doesn`t fall below the
+// minHealthyCount
+func (r *ReconcileWindowsMachine) isAllowedDeletion(machine *mapi.Machine) bool {
+	machinesetName := machine.OwnerReferences[0].Name
```

Comment:
I am not sure if we should use this.
We may want to instead use the `machine.openshift.io/cluster-api-machineset` label on the machine.
The label seems like it is more likely to supported for longer

Reply:
Yes, I tried using the label earlier. However, it doesn`t hold true for our test MachineSets.
e2e Tests -

` MachineSet Name : e2e-windows-machineset-with-windows-label-6ptb`

`machine.openshift.io/cluster-api-machineset=e2e-windows-machineset-with-windows-label-us-west-1a`

Reply:
I just saw this. I think you need to add a guard against `machine.OwnerReferences[0]` not existing at least.

Reply:
>We may want to instead use the machine.openshift.io/cluster-api-machineset label on the machine.
The label seems like it is more likely to supported for longer

I prefer owner references as they are more authoritative. The only problem would be if `MachineSets` are in one namespace and the `Machines` are in another at which point owner refs are not supposed to be added due to the way [GC works](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents).